src/plugins/dashboard/public/application/embeddable/viewport/dashboard_viewport.tsx

```diff
@@ -129,7 +130,7 @@ export class DashboardViewport extends React.Component<DashboardViewportProps, S
             onExitFullScreenMode={this.onExitFullScreenMode}
           />
         )}
-        <DashboardGrid container={container} />
+        <DashboardGrid container={container} PanelComponent={PanelComponent} />
```

Comment:
I feel `dashboard_viewport` should not know about `PanelComponent` as it's really only passing it down to the grid. Would it be better to pass options here and let the grid pull out the PanelComponent?

Reply:
I am not a huge fan of the way it currently works, passing the panelComponent down multiple levels like that. 

That said, I feel like passing the entire options object would be a bit antithetical to this PR, because options could no longer be marked private, and the viewport and grid would have to be aware of the container's options.